R/get_source_expressions.R

```diff
@@ -79,12 +79,12 @@ get_source_expressions <- function(filename, lines = NULL) {
   names(source_expression$lines) <- seq_along(source_expression$lines)
   source_expression$content <- get_content(source_expression$lines)
   parsed_content <- get_source_expression(source_expression, error = function(e) lint_parse_error(e, source_expression))
-  top_level_map <- generate_top_level_map(parsed_content)
 
-  if (inherits(e, "lint") && !nzchar(e$line)) {
+  if (inherits(e, "lint") && (is.na(e$line) || !nzchar(e$line))) {
```

Comment:
As the parser fails out on the last line of the `NA`-blanked version of `lines`, `lint_parse_error_r42()` returns one of these `NA` lines. `nzchar(NA)` is `TRUE`, and `nzchar(NA, keepNA=TRUE)` would fail for `if (NA)`.

Reply:
Since this is the second bug relating to NA lines, can we add a check to `Lint()` to assert that `line` is not `NA`?
This should expedite bug hunts in these cases. WDYT?

Reply:
Hmm, this caused some other tests to fail. Agree it's a good idea, but let's handle it in a follow-up instead.

Reply:
#1450